tests/testthat/test_provenance_spy.R

```diff
@@ -21,11 +21,7 @@ test_that("read.csv captured", {
 
 
 test_that("fread replaced as expected when data.table is loaded after ihme.covid", {
-  expect_false(exists("fread"))
-
-  library(data.table)
-  on.exit(detach("package:data.table", unload = TRUE), add = TRUE)
```

Comment:
What you've removed probably **was** cleaning up after itself properly **assuming** it was run in a separate process

In other words, it cleaned itself up so other tests could run. But also it unloads `data.table` from your global env IFF you didn't isolate the tests from the rest of RStudio

Reply:
You could also make the `on.exit(detach(...))` conditional on data.table **not being loaded** at the beginning of the function.

Then it would never run when you're doing things interactively in RStudio because (presumably) data.table would already be loaded